ARDUINO_UNO form factor and new PinNames for MIMXRT1050 target

notronrj opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

    These changes add support for the new ARDUINO_UNO PinName standard for the MIMXRT1050 target.
    These changes are needed to avoid compiler warnings about ARDUINO form factor being deprecated and allows users of 
    MIMXRT1050 to begin to use the new ARDUINO UNO Pin naming standard release in mbed-os v6.10.

    Files changed: targets.json, PinNames.h

#### Impact of changes <!-- Optional -->

   No real need to change any application code. However using the old Arduino Pin naming standard of D0..D15 etc should now 
   be changed to use ARDUINO_UNO_D0 .. ARDUINO_UNO_D15 etc. Same for the analog pin names.

   Old PinNames such as D0 can continue to be used if desired because of a pin name alias header file:
   `hal/include/hal/PinNameAliases.h`. However, the next version of mbed-os will not include this alias file.
 

   

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->
   None.

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

Commit:
Added Support for ARDUINO_UNO Form Factor on TARGET_MIMXRT1050.
Renamed D0..D15 and A0..A5 to ARDUINO_UNO_D0 etc.

This allows user to use ARDUINO_UNO as the supported_form_factors in targets.json for MIMXRT1050_EVK.

Commit:
Change supported_form_factors from ARDUINO to ARDUINO_UNO to support Pin Name changes for mbed os v6.10.

MarceloSalazar commented on targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT1050/TARGET_EVK/PinNames.h:
```diff
@@ -182,33 +182,33 @@ typedef enum {
+    ARDUINO_UNO_A4 = GPIO_AD_B1_01,
+    ARDUINO_UNO_A5 = GPIO_AD_B1_00,
+
+    I2C_SCL = ARDUINO_UNO_A5,
```
These I2C_SCL / I2C_SDA pins should be removed as already defined here https://github.com/ARMmbed/mbed-os/pull/14617

notronrj replied:
@MarceloSalazar I will remove those lines. After the update and commit to my fork, will that automatically be seen in this PR? Or must a follow some procedure to update this PR with new commits?

Thank you!

notronrj replied:
@MarceloSalazar I will update the PR today. Thanks again.

Commit:
Changed LED definitions to be compliant with Generic Pin Naming standard
Removed I2C_SCL and I2C_SDA from PinName enum as these are now defined in PinNameAliases.h
